cmake/support_pllgpu.cmake

```diff
@@ -0,0 +1,94 @@
+###############################################################################################
+# This file provides macros to process CUDA, CUDAToolkit, cuQuantum and MPI external libraries.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+set(CRAY_MPICH OFF)
+
+# Macro to aid in finding CUDA lib
+macro(findCUDA external_libs)
+    find_package(CUDA REQUIRED)
+    find_package(CUDAToolkit REQUIRED)
+    if(CUDA_FOUND)
+        target_link_libraries(${external_libs} INTERFACE CUDA::cudart)
+    endif()
+endmacro()
+
+# Macro to aid in finding MPI lib
+macro(findMPI external_libs)
+    if(ENABLE_MPI)
+        find_package(MPI REQUIRED)
+        if(MPI_FOUND)
+            message(STATUS "MPI found.")
+        else()
+            message(FATAL_ERROR "MPI is NOT found.")
+        endif()
+
+        string(FIND "${CMAKE_SYSTEM}" "cray" subStrIdx)
+        if(NOT subStrIdx EQUAL -1)
+            if(DEFINED ENV{CRAY_MPICH_VERSION} AND NOT "$ENV{CRAY_MPICH_VERSION}" STREQUAL "none")
+                find_library(MPI_GTL
+                NAMES libmpi_gtl_cuda.so libmpi_gtl_cuda.so.0 libmpi_gtl_cuda.so.0.0.0
+                HINTS ENV CRAY_LD_LIBRARY_PATH
+                ENV LD_LIBRARY_PATH
+                )
+            set(CRAY_MPICH ON)
+            endif()
+        endif()
+
+        target_link_libraries(${external_libs} INTERFACE MPI::MPI_CXX)
+        if($(CRAY_MPICH))
+            target_link_libraries(${external_libs} INTERFACE ${MPI_GTL})
+        endif()
+    endif()
+endmacro()
+
+# Macro to aid in finding cuStateVec lib
+macro(findCustatevec external_libs)
+    find_library(CUSTATEVEC_LIB
+        NAMES   libcustatevec.so.1 custatevec.so.1
```

Comment:
Do the version number change between package increments? How about against the CUDA 11 vs CUDA builds?

Reply:
Not really, there is no difference in the lib name between cu-11 and cu-12.